### PR TITLE
feat: support Query\Builder::toRawSql()

### LINKED ISSUE #119: Add support for toRawSql
Currently, the formatting for boolean is incorrect. 

https://github.com/laravel/framework/blob/87b9c99ef8c3f4acc250716602b4a93defd4665b/src/Illuminate/Database/Connection.php#L1094

### PR DESCRIPTION
**Pull request by taka-oyama**
closes #119

- [x] CHANGELOG

### CODE REVIEW

File: src/Connection.php [hunk 1]
```diff
@@ -420,6 +420,24 @@ public function prepareBindings(array $bindings)
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function escapeString($value)
+    {
+        return str_contains($value, "\n")
+            ? 'r"""' . addcslashes($value, '"') . '"""'
+            : '"' . addcslashes($value, '"') . '"';
```
Comment (taka-oyama) [inline src/Connection.php:438]: probably should escape slashes

```suggestion
            : '"' . addcslashes($value, '"', '\') . '"';
```

Comment (halnique) [inline src/Connection.php:438]: Is there any documentation that can confirm if this process is correct?
I did not understand why you are doing this....

File: tests/Query/BuilderTest.php [hunk 2]
```diff
@@ -867,4 +867,24 @@ public function test_sharedLock(): void
...
+        $this->assertSame($sql, 'select * from `User` where `b` = false', 'false');
+
+        $sql = $conn->table(self::TABLE_NAME_USER)->where('t', 'test')->toRawSql();
+        $this->assertSame($sql, 'select * from `User` where `t` = "test"', 'text');
+
+        $sql = $conn->table(self::TABLE_NAME_USER)->where('t', '"tes\'s"')->toRawSql();
+        $this->assertSame($sql, 'select * from `User` where `t` = "\"tes\'s\""', 'escaped');
+
+        $sql = $conn->table(self::TABLE_NAME_USER)->where('t', "tes\nt")->toRawSql();
+        $this->assertSame($sql, "select * from `User` where `t` = r\"\"\"tes\nt\"\"\"", 'newline');
```
Comment (halnique) [inline tests/Query/BuilderTest.php:888]:
```suggestion

        $sql = $conn->table(self::TABLE_NAME_USER)->where('t', "tes\nt")->toRawSql();
        $this->assertSame($sql, "select * from `User` where `t` = r\"\"\"tes\nt\"\"\"", 'newline');

        $sql = $conn->table(self::TABLE_NAME_USER)->where('t', "t\"e\"s\nt")->toRawSql();
        $this->assertSame($sql, "select * from `User` where `t` = r\"\"\"t\\\"e\\\"s\nt\"\"\"", 'newline with escaped quote');
```

Shouldn't we also test cases where line breaks and escapes are mixed?

File: src/Connection.php (hunk 1 above)
Comment (taka-oyama) [reply src/Connection.php:438]: see https://cloud.google.com/spanner/docs/reference/standard-sql/lexical#quoted_literals

**Review by zeriyoshi — APPROVED** [review]

File: src/Connection.php (hunk 1 above)
Comment (halnique) [reply src/Connection.php:438]: thanks!

**Review by halnique — APPROVED** [review]

**Pull request merged**